src/main/java/se/knowit/bookitevent/controller/EventController.java

```diff
@@ -1,40 +1,96 @@
-import org.springframework.web.server.ResponseStatusException;
-import se.knowit.bookitevent.model.Event;
-import se.knowit.bookitevent.service.EventService;
-
-import java.util.Set;
-import java.util.UUID;
-
-@RestController
-@RequestMapping("/api/v1")
-@CrossOrigin()
-public class EventController {
-
-	private final EventService eventService;
-
-	public EventController(EventService eventService) {
-		this.eventService = eventService;
-	}
-
-	@GetMapping("/events")
-	public Set<Event> findAllEvents() {
-		Set<Event> allEvents = eventService.findAll();
-		if (allEvents.isEmpty()) {
-			throw notFound();
-		}
-		return allEvents;
-	}
-	
-	@GetMapping("/events/{id}")
-	public Event findById(@PathVariable String id) {
-		return eventService.findByEventId(UUID.fromString(id)).orElseThrow(this::notFound);
-	}
-
-	private ResponseStatusException notFound() {
-		return new ResponseStatusException(HttpStatus.NOT_FOUND, "entity not found");
-	}
-}
+package se.knowit.bookitevent.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import se.knowit.bookitevent.dto.EventDTO;
+import se.knowit.bookitevent.dto.EventMapper;
+import se.knowit.bookitevent.model.Event;
+import se.knowit.bookitevent.service.EventService;
+import se.knowit.bookitevent.service.SaveOrUpdateCommand;
+import se.knowit.bookitevent.service.SaveOrUpdateCommand.CommandResult;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.URI;
+import java.util.Set;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toSet;
+import static se.knowit.bookitevent.service.SaveOrUpdateCommand.Outcome.CREATED;
+import static se.knowit.bookitevent.service.SaveOrUpdateCommand.Outcome.UPDATED;
+
+@RestController
+@RequestMapping(EventController.BASE_PATH)
+@CrossOrigin()
+public class EventController {
+	@SuppressWarnings("WeakerAccess")
+	static final String BASE_PATH = "/api/v1/events";
+	private static final URI BASE_URI = URI.create(BASE_PATH + "/");
+    
+    private final EventService eventService;
+
+	public EventController(EventService eventService) {
+		this.eventService = eventService;
+	}
+	
+	@GetMapping({"", "/"})
+	public Set<EventDTO> findAllEvents() {
+		Set<Event> allEvents = eventService.findAll();
+		if (allEvents.isEmpty()) {
+			throw notFound();
+		}
+		EventMapper mapper = new EventMapper();
+		return allEvents.stream().map(mapper::toDTO).collect(toSet());
+	}
+	
+	@GetMapping("/{id}")
+	public EventDTO findById(@PathVariable String id) {
+		Event event = eventService.findByEventId(UUID.fromString(id)).orElseThrow(this::notFound);
+		return new EventMapper().toDTO(event);
+	}
+	
+	@PostMapping(value = {"", "/"}, consumes = "application/json", produces = "application/json")
+	public ResponseEntity<String> createOrUpdateEvent(@RequestBody EventDTO dto) {
```

Comment:
Could maybe rename method to _saveOrUpdateEvent_ so that it follows the same naming as other methods like _saveOrUpdate(EventDTO)_. Alternatively, all occourances of _save_ could be changed to _create_ since the outcome on a save is CREATED.

Reply:
Pushed changes